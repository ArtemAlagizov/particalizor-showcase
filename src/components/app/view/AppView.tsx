import React from "react";
import { IAppView } from "./IAppView";
import { Styled } from "../styled";
import "antd/dist/antd.css";
import MovingPictureShowcase from "../../moving-picture-showcase/container/MovingPictureShowcase";
import ParticleVortexShowcase from "../../particle-vortex-showcase/container/ParticleVortexShowcase";

const AppView: React.FC<IAppView> = ({
  switchShowcase,
  showcase,
  switchButtonTitle
}) => {
  const componentToDisplay =
    showcase === "moving-picture" ? (
      <MovingPictureShowcase />
    ) : (
      <ParticleVortexShowcase />
    );

  return (
    <Styled.TopContainerContainer>
      <Styled.ShowcaseSwitch onClick={switchShowcase}>
        {switchButtonTitle}
      </Styled.ShowcaseSwitch>
      {componentToDisplay}
    </Styled.TopContainerContainer>
  );
};

export default AppView;
