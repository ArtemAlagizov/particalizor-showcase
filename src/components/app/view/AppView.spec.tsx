import React from "react";
import { shallow } from "enzyme";
import AppView from "./AppView";

describe("Given a AppView", () => {
  describe("when rendering", () => {
    it("should match snapshot", () => {
      const component = shallow(
        <AppView
          switchButtonTitle={""}
          showcase={""}
          switchShowcase={jest.fn()}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
