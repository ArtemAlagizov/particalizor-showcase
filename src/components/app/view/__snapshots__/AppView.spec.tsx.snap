// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a AppView when rendering should match snapshot 1`] = `
<styled.div>
  <styled.div
    onClick={[MockFunction]}
  />
  <ParticleVortexShowcase />
</styled.div>
`;
